### PR TITLE
[refactor] Refactor ticker to allow for variable number of character lists

### LINKED ISSUE #67: Add support to prevent ticking of units as suffixes
I'm currently using your library to display instantaneous values of flow from my sensor and what's happening is that the unit (L/m) is a suffix and it's ticking every time the value changes which, ideally, shouldn't happen. Any idea how I'd be dealing with a situation like this?

### PR DESCRIPTION
**Pull request by jinatonic**
This is continuation from #73 , sorry I don't know how PRs work on github.

This is a backward incompatible change in preparation for major release 2.

Breaking changes:

- Changed possible values for `app:ticker_defaultCharacterList`
- Completely recreated `TickerUtils` with new methods
- Removed the need to include `TickerUtils.EMPTY_CHAR` in every character list.
- `setCharacterList` now takes in a vararg of character lists. It’s also been renamed as `setCharacterLists` (plural).
- Supporting variable number of character lists have several advantages.

We don't have to specify the animation behavior from all possible characters.

Characters that do not exist in any character list can be treated differently by the animator.
e.g. always prioritize preserving them via `ACTION_SAME` so they do not get animated.
Can support more custom animation behavior since each character list gets priority over the previous.

e.g. [0, 1, 2] dictates animation behavior from 0 to 2, but what if I want a different behavior
from 1 to 2? I can provide another character list that's something like [1, 3, 2].

Can support wrapping around behavior where 9 -> 0 properly wraps around the list rather than
going all the way down. This is currently not possible because wrap-around in the current
scenario would mean it would animate through all of the non-number characters (e.g. $).

closes #67

### CODE REVIEW

**Comment by jinatonic** [comment]
regarding your comment on `String` -> `char[]` conversion, i think it's fine because most likely when constructing these character lists they would have to do `toCharArray` anyway, and having the API accept string is definitely much easier to read.

File: ticker/src/main/java/com/robinhood/ticker/TickerUtils.java [hunk 1]
```diff
@@ -23,48 +23,13 @@
...
 
-        return charList;
+    public static String provideNumberList() {
+        return "0123456789";
     }
 
-    /**
-     * @return a default ordering for numbers.
-     */
-    public static char[] getDefaultNumberList() {
```
Comment (naturalwarren) [inline ticker/src/main/java/com/robinhood/ticker/TickerUtils.java]: We should update the README to reflect public API changes.

Comment (danh32) [reply ticker/src/main/java/com/robinhood/ticker/TickerUtils.java]: good catch. and add an entry to the changelog.md

File: ticker/src/main/java/com/robinhood/ticker/TickerColumnManager.java [hunk 2]
```diff
@@ -32,37 +35,51 @@
...
     }
 
     /**
      * Tell the column manager the new target text that it should display.
      */
     void setText(char[] text) {
-        if (characterList == null) {
-            throw new IllegalStateException("Need to call setCharacterList(char[]) first.");
+        if (characterLists == null) {
+            throw new IllegalStateException("Need to call setCharacterLists(char[]...) first.");
```
Comment (naturalwarren) [inline ticker/src/main/java/com/robinhood/ticker/TickerColumnManager.java]: String is the argument now :)

**Pull request merged**